Hyperbolic_surface_triangulation_2/include/CGAL/Hyperbolic_surface_triangulation_2.h

```diff
@@ -0,0 +1,721 @@
+     _combinatorial_map.template info<1>(f) = one - (one - cross_ratio_BC_2) / (cross_ratio_BD) ;
+   }
+}
+
+template<class Traits>
+bool Hyperbolic_surface_triangulation_2<Traits>::is_delaunay() const{
+  if (! is_valid()){
+    return false;
+  }
+  return (pick_edge_to_flip() == nullptr);
+}
+
+template<class Traits>
+int Hyperbolic_surface_triangulation_2<Traits>::make_delaunay(){
+  int number_of_flips_done = 0;
+
+  Dart_handle edge_to_flip = pick_edge_to_flip();
+  while (edge_to_flip != nullptr){
+    flip(edge_to_flip);
+    edge_to_flip = pick_edge_to_flip();
+    number_of_flips_done++;
+  }
+
+  return number_of_flips_done;
+}
+
+
+template<class Traits>
+std::vector<std::tuple<typename Hyperbolic_surface_triangulation_2<Traits>::Dart_const_handle, typename Hyperbolic_surface_triangulation_2<Traits>::Point, typename Hyperbolic_surface_triangulation_2<Traits>::Point, typename Hyperbolic_surface_triangulation_2<Traits>::Point>> Hyperbolic_surface_triangulation_2<Traits>::lift(bool center) const{
+  std::vector<std::tuple<Dart_const_handle,Point,Point,Point>> realizations;
+
+  size_t visited_darts_mark = _combinatorial_map.get_new_mark();
+  _combinatorial_map.unmark_all(visited_darts_mark);
+
+  struct Compare {
+    bool operator()(std::pair<Dart_const_handle,double> const & x, std::pair<Dart_const_handle,double> const & y) {
+        return x.second > y.second;
+    }
+  };
+  std::priority_queue<std::pair<Dart_const_handle,double>, std::vector<std::pair<Dart_const_handle,double>>, Compare> queue;
+
+  std::map<Dart_const_handle, Point> positions;
+
+  Dart_const_range darts = _combinatorial_map.darts();
+
+  _combinatorial_map.mark(_anchor.dart, visited_darts_mark);
+  _combinatorial_map.mark(const_ccw(_anchor.dart), visited_darts_mark);
+  _combinatorial_map.mark(const_cw(_anchor.dart), visited_darts_mark);
+
+  if (center){
+    Isometry center_the_drawing = hyperbolic_translation<Traits>(_anchor.vertices[0]);
+    positions[_anchor.dart] = center_the_drawing.evaluate(_anchor.vertices[0]);
+    positions[const_ccw(_anchor.dart)] = center_the_drawing.evaluate(_anchor.vertices[1]);
+    positions[const_cw(_anchor.dart)] = center_the_drawing.evaluate(_anchor.vertices[2]);
+  } else {
+    positions[_anchor.dart] = _anchor.vertices[0];
+    positions[const_ccw(_anchor.dart)] = _anchor.vertices[1];
+    positions[const_cw(_anchor.dart)] = _anchor.vertices[2];
+  }
+
+  std::tuple<Dart_const_handle,Point,Point,Point> value = std::make_tuple(_anchor.dart, positions[_anchor.dart], positions[const_ccw(_anchor.dart)], positions[const_cw(_anchor.dart)]);
+  realizations.push_back(value);
+
+  ComplexNumber anchor_z0 (_anchor.vertices[0].x(), _anchor.vertices[0].y());
+  ComplexNumber anchor_z1 (_anchor.vertices[1].x(), _anchor.vertices[1].y());
+  ComplexNumber anchor_z2 (_anchor.vertices[2].x(), _anchor.vertices[2].y());
+
+  double weight_of_anchor_dart = CGAL::to_double(anchor_z0.squared_modulus() + anchor_z1.squared_modulus());
```

Comment:
Spotting a `double` without understanding what the function is about immediately leads to the question if this could lead to wrong results.

Reply:
The weights instantiated via a to_double are just used here for a heuristic algorithm to deal with a priority queue during the lift, the result of the lift is exact but the order in which the lift is processed does not need more than a heuristic.
@loic-dubois : anything to add?